### PR TITLE
170 add show methods

### PR DESCRIPTION
**Pull request by darkgoldenrod-cherry**

### CODE REVIEW

File: R/te_outcome_model.R [hunk 1]
```diff
@@ -36,3 +36,38 @@ setClass(
...
+setMethod(
+  "show",
+  c(object = "te_stats_glm_logit_outcome_fitted"),
+  function(object) {
+    catn("te_stats_glm_logit_outcome_fitted object")
+    catn("")
+    catn("Model:")
+    show(object@model$model)
+    catn("")
+    # matrix @model$vcov omitted
```
Comment (gravesti) [inline R/te_outcome_model.R:66]: I think we can remove this since we have the model summary

**Comment by gravesti** [comment]
Looks like we might need to extend our classes or the show method implementations to know when the outcome model is not set yet, and also when the outcome model is not fitted yet. @darkgoldenrod-cherry

**Comment by darkgoldenrod-cherry** [comment]
I wrote a small helper function to get a dataframe of all existing classes and whether a show method exists for them or not. This might be helpful down the line for planning.

```
                               class show_method_exists
1                            te_data               TRUE
2                      te_data_unset              FALSE
3                       te_datastore              FALSE
4                   te_datastore_csv              FALSE
5             te_datastore_datatable              FALSE
6                te_datastore_duckdb              FALSE
7                       te_expansion              FALSE
8                 te_expansion_unset              FALSE
9                    te_model_fitter              FALSE
10                   te_outcome_data              FALSE
11                 te_outcome_fitted              FALSE
12                  te_outcome_model               TRUE
13                te_stats_glm_logit              FALSE
14 te_stats_glm_logit_outcome_fitted               TRUE
15                 te_weights_fitted               TRUE
16                   te_weights_spec               TRUE
17                  te_weights_unset               TRUE
18                    trial_sequence               TRUE
19                 trial_sequence_AT              FALSE
20                trial_sequence_ITT              FALSE
21                 trial_sequence_PP              FALSE
```

**Comment by darkgoldenrod-cherry** [comment]
Now it works partially, though te_outcome@model and te_outcome@summary get printed when they're empty. 
  - Should I make a new class again or should I go for an `if`statement?

Also for some strange reason that I can't figure out it prints show(object@switch_weights) twice, the first time where it should be and the second time at the very end like it was specified with the child class methods that were deleted (also according to my helper method there isn't currently a show method for any of the child classes). I restarted R, reloaded the package, it didn't help. If I just run the last line of the show method, the switch_weights don't show up, so the problem is not in there.

```
Trial Sequence Object 
Estimand: Per-protocol 
Data 
N: 321 observations from 89 patients 
Key: <id>
        id period treatment    x1           x2    x3        x4   age      age_s outcome censored eligible time_of_event
     <int>  <int>     <num> <num>        <num> <int>     <num> <num>      <num>   <num>    <int>    <num>         <num>
  1:     1      0         1     1  1.146148362     0 0.7342030    36 0.08333333       0        0        1          9999
  2:     1      1         1     1  0.002200337     0 0.7342030    37 0.16666667       0        0        0          9999
 ---                                                                                                                   
320:    99      1         1     0 -1.106480738     1 0.5752681    66 2.58333333       0        0        0             7
321:    99      2         0     0  1.650478074     1 0.5752681    67 2.66666667       0        0        0             7
      first  am_1  cumA switch regime_start time_on_regime eligible0 eligible1
     <lgcl> <num> <num>  <num>        <int>          <num>     <num>     <num>
  1:   TRUE     0     1      0            0              0         1         0
  2:  FALSE     1     2      0            0              1         0         1
 ---                                                                          
320:  FALSE     1     2      0            0              1         0         1
321:  FALSE     1     2      1            2              2         0         1
 
IPW for informative censoring: 
 - No weight model specified 
 
IPW for treatment switch censoring:    <----------------------
 - No weight model specified
 
TE Outcome Model Object 
Formula: outcome ~ assigned_treatment + followup_time + I(followup_time^2) + trial_period + I(trial_period^2) 
Treatment_var: assigned_treatment 
Adjustment_vars:  
 
An object of class "te_outcome_fitted"
Slot "model":
list()

Slot "summary":
list()

 
IPW for treatment switch censoring:    <----------------------
 - No weight model specified
```

File: R/trial_sequence.R [hunk 2]
```diff
@@ -112,28 +113,15 @@ setMethod(
...
+    if (.hasSlot(object, "switch_weights")) {
+      catn("")
+      catn("IPW for treatment switch censoring:")
+      show(object@switch_weights)
+    }
+    # expansion
     catn("")
-    catn("IPW for treatment switch censoring:")
-    show(object@switch_weights)
+    show(object@outcome_model)
```
Comment (gravesti) [inline R/trial_sequence.R:123]: How about a heading here?
```suggestion
    catn("Outcome model:")
    show(object@outcome_model)
```

File: R/te_outcome_model.R [hunk 3]
```diff
@@ -30,9 +30,51 @@ setClass("te_outcome_model",
...
+    show(object@fitted)
+    # outcome_data?
+  }
+)
+
+# show method for te_outcome_model_unset -----
+setMethod(
+  "show",
+  c(object = "te_outcome_model_unset"),
+  function(object) cat("Outcome model not specified. See set_outcome_model()")
```
Comment (gravesti) [inline R/te_outcome_model.R:63]:
```suggestion
  function(object) cat(" - Outcome model not specified. Use set_outcome_model()")
```

File: R/te_outcome_model.R [hunk 4]
```diff
@@ -30,9 +30,51 @@ setClass("te_outcome_model",
...
+)
+
+# show method for te_outcome_fitted -----
+setMethod(
+  "show",
+  c(object = "te_outcome_fitted"),
+  function(object) {
+    if (length(object@summary)) {
+      catn("TE Outcome Fitted Object")
```
Comment (gravesti) [inline R/te_outcome_model.R:72]: Maybe that heading isn't interesting to the user
```suggestion
```

File: R/te_outcome_model.R [hunk 5]
```diff
@@ -30,9 +30,51 @@ setClass("te_outcome_model",
...
+      catn("Model Summary:")
+      catn("")
+      show(as.data.frame(object@summary$tidy))
+      catn("")
+      show(as.data.frame(object@summary$glance))
+    }
```
Comment (gravesti) [inline R/te_outcome_model.R:78]:
```suggestion
    } else {
    Use fit_msm() to fit the outcome model
    }
```

**Review by gravesti — COMMENTED** [review]
Really nice @darkgoldenrod-cherry 
A few small suggestions

**Comment by gravesti** [comment]
@darkgoldenrod-cherry Can we have one more snapshot where we see the printed model summary? eg add a test_snapshot to one of the tests in test-modelling.R

File: R/te_outcome_model.R [hunk 6]
```diff
@@ -30,9 +30,52 @@ setClass("te_outcome_model",
...
+
+# show method for te_outcome_fitted -----
+setMethod(
+  "show",
+  c(object = "te_outcome_fitted"),
+  function(object) {
+    if (length(object@summary)) {
+      catn("Model Summary:")
+      catn("")
+      show(as.data.frame(object@summary$tidy))
```
Comment (gravesti) [inline R/te_outcome_model.R:74]:
```suggestion
      print.data.frame(as.data.frame(trial_itt@outcome_model@fitted@summary$tidy), row.names = FALSE, digits = 2, right = FALSE)
```

File: R/te_outcome_model.R [hunk 7]
```diff
@@ -30,9 +30,52 @@ setClass("te_outcome_model",
...
+      catn("")
+      show(as.data.frame(object@summary$glance))
```
Comment (gravesti) [inline R/te_outcome_model.R:76]:
```suggestion
      print.data.frame(as.data.frame(trial_itt@outcome_model@fitted@summary$glance), row.names = FALSE, digits = 3, right = FALSE)
```

**Review by gravesti — COMMENTED** [review]
One more suggestion on the data frames :)

File: R/te_outcome_model.R (hunk 7 above)
Comment (darkgoldenrod-cherry) [reply R/te_outcome_model.R:76]: This is a method for te_outcome_fitted, wouldn't replacing `object` with `trial_itt@outcome_model@fitted` break it? @gravesti

Comment (gravesti) [reply R/te_outcome_model.R:76]: oops, sorry that was the code I was testing with.

**Review by gravesti — APPROVED** [review]
Thanks @darkgoldenrod-cherry !

**Comment by darkgoldenrod-cherry** [comment]
I don't know what to do with this one failing check, everything works on my PC. Any Ideas? @gravesti

**Comment by gravesti** [comment]
@darkgoldenrod-cherry It turns out the example we used was not so good, so it was numerically unstable so we saw the tiny differences between platforms being blown up. I have changed the test so it is more stable

**Comment by darkgoldenrod-cherry** [comment]
Ah, okay. Thank you

**Pull request merged**